File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/03_Advantages/03_Flexibility/javascript/main.js
const { Car, Bike, Truck } = require('../vehicle');

let vehicle;

// Flexible: Dynamically assign different types of vehicles
vehicle = new Car();
vehicle.start(); // Output: Starting the car

vehicle = new Bike();
vehicle.start(); // Output: Starting the bike

vehicle = new Truck();
vehicle.start(); // Output: Starting the truck

/*
Output:
Starting the car
Starting the bike
Starting the truck
*/

File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/03_Advantages/03_Flexibility/python/main.py
import sys
sys.path.append('..')
from vehicle import Car, Bike, Truck

# Flexible: Dynamically assign different types of vehicles
vehicle = Car()
vehicle.start()  # Output: Starting the car

vehicle = Bike()
vehicle.start()  # Output: Starting the bike

vehicle = Truck()
vehicle.start()  # Output: Starting the truck

"""
Output:
Starting the car
Starting the bike
Starting the truck
"""

